transport/server.cc

```diff
@@ -1275,6 +1278,15 @@ std::unique_ptr<cql_server::response> cql_server::connection::make_function_fail
     return response;
 }
 
+std::unique_ptr<cql_server::response> cql_server::connection::make_rate_limit_error(int16_t stream, exceptions::exception_code err, sstring msg, const tracing::trace_state_ptr& tr_state, const service::client_state& client_state) const
+{
+    if (!client_state.is_protocol_extension_set(cql_protocol_extension::RATE_LIMIT_ERROR)) {
+        return make_error(stream, exceptions::exception_code::CONFIG_ERROR, std::move(msg), tr_state);
+    }
+
```

Comment:
Very good! This both takes care of compatibility and gives fine-grained information for updated drivers.